forum/views/pins.py

```diff
@@ -12,6 +12,28 @@
 from forum.serializers.thread import ThreadSerializer
 
 
+def pin_unpin_thread(
+    user_id: str,
+    thread_id: str,
+    action: str,
+) -> dict[str, Any]:
+    """Pin or Unpin  a thread."""
+    try:
+        serialized_data: dict[str, Any] = handle_pin_unpin_thread_request(
+            user_id, thread_id, action, ThreadSerializer
+        )
+    except ValueError as e:
+        return {
+            "text": str(e),
+            "status_code": status.HTTP_400_BAD_REQUEST,
```

Comment:
As discussed in the [corresponding edx-platform PR](https://github.com/edly-io/edx-platform/pull/580#pullrequestreview-2300338139), I suggest we raise issues directly from here instead of pretending that we are still using a REST API.